tests/tests.go

```diff
@@ -44,13 +44,12 @@ func StartServerWithContext(t testing.TB, cfg *serverconfig.Config, serverCtx *r
 	go func() {
 		serverDone <- serverCtx.Run(ctx, cfg)
 	}()
-
-	testutils.EnsureServiceHealthy(t, cfg.GRPC.Addr, cfg.HTTP.Addr, nil, false)
```

Comment:
if this line returns an error and the test exits, the `t.Cleanup` below will never run 😅